tallycash/extension
# NFTs background

Ref #2619
### What

- This PR introduces NFTs service, database and redux slice with basic functionality:
  - fetching collections by account and tracked networks, saving them to database
  - fetching and saving nfts from given collections
  - caching collections in redux store
- Adds basic types: NFT, NFTCollection
- Adds new nfts providers map
- Handles fetching collections and NFTs from SimpleHash and POAP APIs

Based on the `SUPPORT_NFT_TAB`:

- if flag is `true`: create service and database, create `nfts` and `collections` tables, initialize redux slice based on the database contents
- if flag is `false`: create service but don't connect it to other services, create database but don't create any tables, don't initialize redux slice at all

### Testing

- with `SUPPORT_NFT_TAB = false` 
  - reload extension, check if there is no database `tally/nfts` visible in DevTools Application >  IndexedDB
  - check if no `nftsUpdate` redux slide was created using script:  
```javascript
chrome.storage.local.get(["state"], ({state: stateString}) => {
    const state = JSON.parse(stateString, null, 2)
    console.log(state)
})
```
- with `SUPPORT_NFT_TAB = true` 
  - reload extension, redux slice should be created and collections should be fetched for accounts that are added (check db and redux slice)

Latest build: [extension-builds-2642](https://github.com/tallycash/extension/suites/9448736594/artifacts/446992982) (as of Tue, 22 Nov 2022 16:13:38 GMT).
